core/src/main/java/nl/nn/adapterframework/doc/model/FrankDocModel.java

```diff
@@ -1,46 +1,136 @@
+			throw new RuntimeException(String.format("Cannot find resource on the classpath: [%s]", path));
+		}
+		try {
+			XmlUtils.parseXml(resource.asInputSource(), new Handler(path));
+		}
+		catch(IOException e) {
+			throw new RuntimeException(String.format("An IOException occurred while parsing XML from [%s]", path), e);
+		}
+		catch(SAXException e) {
+			throw new RuntimeException(String.format("A SAXException occurred while parsing XML from [%s]", path), e);
+		}
+	}
+
+	private class Handler extends DigesterRulesHandler {
+		private final String path;
+
+		Handler(String path) {
+			this.path = path;
+		}
+
+		@Override
+		protected void handle(DigesterRule rule) throws SAXException {
+			String pattern = rule.getPattern();
+			StringTokenizer tokenizer = new StringTokenizer(pattern, "/");
+			String syntax1Name = null;
+			while(tokenizer.hasMoreElements()) {
+				String token = tokenizer.nextToken();
+				if(!"*".equals(token)) {
+					syntax1Name = token;
+				}
+			}
+			if(StringUtils.isNotEmpty(rule.getRegisterMethod())) {
+				add(rule.getRegisterMethod(), syntax1Name);
+			}			
+		}
+
+		private void add(String registerMethod, String syntax1Name) throws SAXException {
+			ConfigChildSetterDescriptor item = new ConfigChildSetterDescriptor(registerMethod, syntax1Name);
+			if(configChildDescriptors.containsKey(item.getMethodName())) {
+				log.warn(String.format("In digester rules [%s], duplicate method name [%s]", path, registerMethod));
+			} else {
+				configChildDescriptors.put(item.getMethodName(), item);
+			}
+		}
+	}
+
+	public ElementType findOrCreateElementType(Class<?> clazz) {
+		if(allTypes.containsKey(clazz.getName())) {
+			return allTypes.get(clazz.getName());
+		}
+		final ElementType result = new ElementType(clazz);
+		// If a containing FrankElement contains the type being created, we do not
+		// want recursion.
+		allTypes.put(result.getFullName(), result);
+		if(clazz.isInterface()) {
+			Utils.getSpringBeans(clazz.getName()).stream()
+					.map(b -> b.getClazz())
+					.map(cl -> findOrCreateFrankElement(cl))
+					.forEach(result::addMember);
+		} else {
+			result.addMember(findOrCreateFrankElement(clazz));
+		}
+		return result;
+	}
+
+	public boolean hasType(String typeName) {
+		return allTypes.containsKey(typeName);
+	}
+
 	public FrankElement findOrCreateFrankElement(Class<?> clazz) {
 		if(allElements.containsKey(clazz.getName())) {
 			return allElements.get(clazz.getName());
 		}
 		Class<?> superClass = clazz.getSuperclass();
 		FrankElement parent = superClass == null ? null : findOrCreateFrankElement(superClass);
-		FrankElement current = new FrankElement(clazz, parent);
-		current.setAttributes(createAttributes(clazz.getDeclaredMethods(), current));
-		allElements.put(current.getFullName(), current);
-		return current;
+		if(allElements.containsKey(clazz.getName())) {
+			// This can happen when element C inherits from element P, while P can have
+			// C as a configuration child.
```

Comment:
Met 'this can happen' bedoel je denk ik: dat dit FrankElement al aangemaakt is als gevolg van het aanmaken van de parent class. Als dat zo is, dan moet je dat concreter zeggen.

Reply:
Akkoord, ga ik verbeteren.

Reply:
Ook over deze if():
Als je problemen met circular references wil voorkomen (bijvoorbeeld door een IbisDocRef van een class naar een child), dan moet je niet deze if doen, maar het element al aanmaken en in de lijst opnemen vóór dat je findOrCreateFrankElement(superClass) aanroept. 
De if() die je nu hebt gaat je er niet bij helpen, die komt te laat. Ik denk dus dat hij weg kan.

Reply:
Gedaan.